### ISSUE TITLE
Entire line selection on clicking the index number/ 

### ISSUE THREAD

**Issue by AnkurIngale** [processing/p5.js-web-editor#895]
#### Nature of issue?

- New feature request

#### Details about the bug: 
Like other editors, there should be a feature to select the entire line on clicking the index number of that line.
- Web browser and version:  On Chrome
- Operating System: MacOSX
- Steps to reproduce this bug:

#### New feature details:
This can be done by setting an onclick method and indexing the lines to their respective numbers and highlighting it. What do you think @catarak ?

**Comment by catarak** [comment]
wow, i didn't even know you could do this! i wonder if there's an addon to codemirror that makes this easy to implement.

**Comment by catarak** [comment]
to me, this issue is not high priority, but it doesn't seem like it overcomplicates the editor or gets in the way. implement it if you think is important!

**Comment by AnkurIngale** [comment]
@catarak  https://codemirror.net/doc/manual.html#addLineClass If we apply this and select the code, this should work. What do you think?

**Comment by AnkurIngale** [comment]
@catarak I have made this feature but there is a problem I want to know about. Are all the edits in the editor possible only after clicking on them? I mean it specifically for the template code at the start. Because once clicked then the selection works fine on double click.

**Comment by catarak** [comment]
which PR? it's not linked to this issue so i can't find it.

**Comment by hellonearthis** [comment]
Having the text caret/cursor on a line without selecting any text lets you use ctrl-c and ctrl-x to copy and cut the entire line BTW.

**Comment by AnkurIngale** [comment]
@catarak #907 This one.

**Comment by AnkurIngale** [comment]
@hellonearthis Yeah, right! Is it documented somewhere?

**Comment by AnkurIngale** [comment]
I tried the following code but on a discussion, we found that it is not an appropriate solution.
<img width="562" alt="Screenshot 2019-04-04 at 12 55 05 AM" src="https://user-images.githubusercontent.com/33569473/55507042-7de70d00-5674-11e9-870d-8b4c50e3e9da.png">
You can go through our discussion here: https://github.com/processing/p5.js-web-editor/pull/984

**Comment by AnkurIngale** [comment]
I then planned on making a line active (grey background) on clicking on respective line number. But then if I couldn't differentiate between fold marker and line number click as of now. Here's a snippet of the code:
<img width="552" alt="Screenshot 2019-04-04 at 12 54 27 AM" src="https://user-images.githubusercontent.com/33569473/55507217-cbfc1080-5674-11e9-912b-2d16f1668328.png">
So we decided not to have this as well and closed the PR. @siddhant1 @plxity I need your reviews on it and a possible solution for the feature.

**Comment by plxity** [comment]
@AnkurIngale The features seem nice. I am thinking about this issue. I will let you know if I get a proper solution or reference.

**Comment by catarak** [comment]
Seems like getting this to work is surprisingly hard? This is pretty low priority, and given that it doesn't work well with CodeMirror, i'm going to close this unless it's something that CodeMirror adds as part of its API.
